tests/foreman/cli/test_ansible.py

```diff
@@ -677,3 +679,273 @@ def _finalize():
+        if auth_type == 'admin':
+            login = gen_string('alpha')
+            user = target_sat.api.User(
+                admin=True,
+                login=login,
+                password=password,
+                organization=[module_org],
+                location=[module_location],
+            ).create()
+        else:
+            login = gen_string('alpha')
+            role = target_sat.api.Role(
+                name=gen_string('alpha'), location=[module_location], organization=[module_org]
+            ).create()
+            for perm in ['view_hosts', 'view_hostgroups', 'view_facts']:
+                permission = target_sat.api.Permission().search(query={'search': f'name={perm}'})
+                target_sat.api.Filter(permission=permission, role=role).create()
+            user = target_sat.api.User(
+                role=[role],
+                admin=False,
+                login=login,
+                password=password,
+                organization=[module_org],
+                location=[module_location],
+            ).create()
+            target_sat.cli.User.add_role(
+                {'login': user.login, 'role': 'Ansible Tower Inventory Reader'}
+            )
+        result = rhel_contenthost.register(
+            module_org,
+            module_location,
+            module_ak_with_cv.name,
+            target_sat,
+            auth_username=user.login,
+            auth_password=password,
+            force=True,
+        )
+        assert result.status == 0, f'Failed to register host: {result.stderr}'
+
+        if is_open('SAT-28613') and auth_type == 'non-admin':
+            host = rhel_contenthost.nailgun_host
+            host.location = module_location
+            host.update(['location'])
+
+        # Find the Satellite credentials in AAP and update it for target_sat.hostname and user credentials
+        creds_list = aap_client.get(
+            '/api/v2/credentials/', query_parameters=f'name={creds_name}'
+        ).json()
+        new_creds = {
+            'inputs': {
+                'host': f'https://{target_sat.hostname}',
+                'username': login,
+                'password': settings.server.admin_password,
+            }
+        }
+        response = aap_client.patch(
+            f'/api/v2/credentials/{creds_list["results"][0]["id"]}/', json=new_creds
+        )
+        assert response.ok
+
+        # Find the Satellite inventory in AAP and update it for target_sat and user credentials
+        inv_list = aap_client.get(
+            '/api/v2/inventories/', query_parameters=f'name={inventory_name}'
+        ).json()
+        inv_source_list = aap_client.get(
+            f'/api/v2/inventories/{inv_list["results"][0]["id"]}/inventory_sources/'
+        ).json()
+        sync_response = aap_client.post(
+            f'/api/v2/inventory_sources/{inv_source_list["results"][0]["id"]}/update/'
+        )
+        assert sync_response.ok
+        sleep(180)
```

Comment:
I see you imported `wait-for` lib. Why not use it here? Use of static timeouts is discouraged.

Reply:
Yes, I agree, I was hoping not to use wait_for earlier here, as it would make the code less readable, but I updated, let's see how it goes :)